images/win/scripts/ImageHelpers/InstallHelpers.ps1

```diff
@@ -687,3 +704,81 @@ function Get-HashFromGitHubReleaseBody {
     }
     return $result
 }
+
+function Use-ChecksumСomparison {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$LocalFileHash,
+        [Parameter(Mandatory=$true)]
+        [string]$DistributorFileHash
+    )
+
+    Write-Verbose "Performing checksum verification"
+
+    if ($LocalFileHash -ne $DistributorFileHash) {
+        throw "Checksum verification failed. Expected hash: $DistributorFileHash; Actual hash: $LocalFileHash."
+    } else {
+        Write-Verbose "Checksum verification passed"
+    }
+}
+
+function Get-HashFromGitHubReleaseBody {
+    param (
+        [string]$RepoOwner,
+        [string]$RepoName,
+        [Parameter(Mandatory=$true)]
+        [string]$FileName,
+        [string]$Url,
+        [string]$Version = "latest",
+        [boolean]$IsPrerelease = $false,
+        [int]$SearchInCount = 100,
+        [string]$Delimiter = '|',
+        [int]$WordNumber = 1
+    )
+
+    if ($Url) {
+        $releaseUrl = $Url
+    } else {
+        if ($Version -eq "latest") {
+            $releaseUrl = "https://api.github.com/repos/${RepoOwner}/${RepoName}/releases/latest"
+        } else {
+            $json = Invoke-RestMethod -Uri "https://api.github.com/repos/${RepoOwner}/${RepoName}/releases?per_page=${SearchInCount}"
+            $tags = $json.Where{ $_.prerelease -eq $IsPrerelease }.tag_name
+            $tag = $tags -match $Version
+            if (-not $tag) {
+                throw "Failed to get a tag name for version $Version."
+            }
+            $releaseUrl = "https://api.github.com/repos/${RepoOwner}/${RepoName}/releases/tag/$tag"
+        }
+    }
+    $body = (Invoke-RestMethod -Uri $releaseUrl).body -replace('`', "") -join "`n"
```

Comment:
It's not clear why do we need to remove backticks from the request body and why do we join response lines first and splitting them afterwards.